src/common/misc/ListModel.ts

```diff
@@ -577,9 +468,120 @@ export class ListModel<ElementType extends ListElement> {
 			this.updateState({ loadingStatus: ListLoadingState.ConnectionLost })
 		}
 	}
+
+	waitLoad(what: () => any): Promise<any> {
+		return settledThen(this.loading, what)
+	}
+
+	addToLoadedEntities(entity: ElementType) {
```

Comment:
should maybe not refer to `entities` in the names since we have specific meaning for it